tests/lint/git-black.sh

```diff
@@ -46,16 +46,20 @@ fi
 echo "Version Information: $(black --version)"
 
 # Compute Python files which changed to compare.
-IFS=$'\n' read -a FILES -d'\n' < <(git diff --name-only HEAD $1 -- "*.py" "*.pyi") || true
-echo "read returned $?"
+IFS=$'\n' read -a FILES -d'\n' < <(git diff --name-only $1 -- "*.py" "*.pyi") || true
+echo "Read returned $?"
+if [ -z ${FILES+x} ]; then
+    echo "No changes in Python files"
+    exit 0
+fi
 echo "Files: $FILES"
 
 if [[ ${INPLACE_FORMAT} -eq 1 ]]; then
-    echo "Running black on Python files against revision" $1):
+    echo "Running black on Python files against revision" $1:
     CMD=( "black" "${FILES[@]}" )
     echo "${CMD[@]}"
     "${CMD[@]}"
 else
     echo "Running black in checking mode"
-    black --diff --check
+    black --diff --check ${FILES[@]}
```

Comment:
I was intentionally having this check the entire repo, not sure if we want to be incremental when doing the full check vs. being incremental when we are formatting. cc @tqchen

Reply:
It looks like I have to provide something; otherwise it will just pop the following message in all cases.
```
ubuntu@~/tvm> black --diff --check
No Path provided. Nothing to do 😴
```
Black version 20.8b1. Not sure if this is due to my local environment issue or something else.

Reply:
I think incremental is fine in most cases